suitcase/fields.py

```diff
@@ -831,6 +832,12 @@ def unpack(self, data, **kwargs):
         self._value = self.substructure()
         return self._value.unpack(data, **kwargs)
 
+    def setval(self, value):
+        if isinstance(value, suitcase.structure.Structure):
```

Comment:
Perhaps I am thinking slowly today.  Is there some instance where this would not be the case?  I think this change makes sense, but I'm wondering if it might make more sense to do something like this instead:

```
def setval(self, value):
    if not isinstance(value, self.substructure):
        raise TypeError("SubstructureField expected %r instance, got %r" % (self.substructure, type(value)))
    else:
        self._value = value
```